tools/generator/SourceWriters/Attributes/ObsoleteAttr.cs

```diff
@@ -24,22 +20,20 @@ public ObsoleteAttr (string message = null, bool isError = false)
 
 		public override void WriteAttribute (CodeWriter writer)
 		{
-			var attr_name = WriteAttributeSuffix ? "ObsoleteAttribute" : "Obsolete";
+			var parts = new List<string> ();
 
-			if (WriteGlobal)
-				attr_name = "global::System." + attr_name;
-
-			if (Message is null && !WriteEmptyString && !IsError) {
-				writer.WriteLine ($"[{attr_name}]");
-				return;
-			}
-
-			writer.Write ($"[{attr_name} ({(NoAtSign ? "" : "@")}\"{Message}\"");
+			if (Message != null)
+				parts.Add ($"@\"{Message}\"");
 
 			if (IsError)
-				writer.Write (", error: true");
+				parts.Add ("error: true");
+
+			var content = string.Join (", ", parts.ToArray ());
 
-			writer.WriteLine (")]");
+			if (content.HasValue ())
+				writer.WriteLine ($"[global::System.Obsolete ({content})]");
```

Comment:
A concern with this approach is that it could result in invalid code: if `Message` is null and `IsError` is true, then we would attempt to emit `[Obsolete(error:true)]`, which will fail to compile as there is no `Obsolete(bool)` [constructor](https://docs.microsoft.com/en-us/dotnet/api/system.obsoleteattribute.-ctor?view=net-6.0).

Is it *possible* for `Message` to be null and `IsError` to be true?

One one level, "no", because `@deprecated` is "never" `null`.

Except when we only pass null:

```
% git grep 'SourceWriterExtensions.AddObsolete.*null'    
tools/generator/SourceWriters/MethodCallback.cs:                        SourceWriterExtensions.AddObsolete (Attributes, null, forceDeprecate: !string.IsNullOrWhiteSpace (method.Deprecated));
tools/generator/SourceWriters/MethodCallback.cs:                        SourceWriterExtensions.AddObsolete (Attributes, null, forceDeprecate: !string.IsNullOrWhiteSpace (method.Deprecated))
```

These callsites don't use `isError:true`, so we should be fine.